taskchampion/src/taskdb/mod.rs

```diff
@@ -31,6 +34,54 @@ impl TaskDb {
         TaskDb::new(Box::new(InMemoryStorage::new()))
     }
 
+    /// Apply `operations` to the dataabase, in a single transaction.
+    ///
+    /// The resulting operations will be appended to the list of local operations, and the
+    /// set of tasks will be updated accordingly.
+    ///
+    /// Any operations for which `add_to_working_set` returns true will cause the relevant
+    /// task to be added to the working set.
+    pub(crate) fn commit_operations<F>(
+        &mut self,
+        operations: Operations,
+        add_to_working_set: F,
+    ) -> Result<()>
+    where
+        F: Fn(&Operation) -> bool,
+    {
+        let mut txn = self.storage.txn()?;
+        apply::apply_operations(txn.as_mut(), &operations)?;
+
+        // Calculate the task(s) to add to the working set.
+        let mut to_add = Vec::new();
+        for operation in &operations {
+            if add_to_working_set(operation) {
+                if let Some(uuid) = match operation {
+                    Operation::Create { uuid } => Some(*uuid),
+                    Operation::Update { uuid, .. } => Some(*uuid),
+                    Operation::Delete { uuid, .. } => Some(*uuid),
+                    _ => None,
+                } {
+                    to_add.push(uuid);
+                }
```

Comment:
I'm not sure why it would be invalid? The match expression has type `Option<Uuid>` and the `if let Some(uuid) = .. { }` matches against that type. But I think I can use alternation in a single match expression to say the same thing more clearly.

Reply:
Thanks, I do find that more clear. After much squinting my eyes can now parse the original expression but I probably wouldn't be able to do it again next week.